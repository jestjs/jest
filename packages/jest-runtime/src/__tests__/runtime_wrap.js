/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

let createRuntime;

describe('Runtime', () => {
  beforeEach(() => {
    createRuntime = require('createRuntime');
  });

  describe('wrapCodeInModuleWrapper', () => {
    it('generates the correct args for the module wrapper', async () => {
      const runtime = await createRuntime(__filename);

      expect(
        runtime.wrapCodeInModuleWrapper('module.exports = "Hello!"'),
      ).toMatchSnapshot();
    });

    it('injects "extra globals"', async () => {
      const runtime = await createRuntime(__filename, {
        sandboxInjectedGlobals: ['Math'],
      });

      expect(
        runtime.wrapCodeInModuleWrapper('module.exports = "Hello!"'),
      ).toMatchSnapshot();
    });
  });
});
