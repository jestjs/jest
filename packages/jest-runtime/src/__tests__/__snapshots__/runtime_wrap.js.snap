// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Runtime wrapCodeInModuleWrapper generates the correct args for the module wrapper 1`] = `
"({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){module.exports = "Hello!"
}});"
`;

exports[`Runtime wrapCodeInModuleWrapper injects "extra globals" 1`] = `
"({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest,Math){module.exports = "Hello!"
}});"
`;
